### PR TITLE
fix: put code inside anchors for IDL expansions

### PR DESCRIPTION
**Pull request by marcoscaceres**

### CODE REVIEW

File: src/core/link-to-dfn.js [hunk 1]
```diff
@@ -212,7 +212,10 @@ function wrapAsCode(ant, dfn) {
   // only add code to IDL when the definition matches
   const term = ant.textContent.trim();
   const isIDL = dfn.dataset.hasOwnProperty("idl");
-  const needsCode = shouldWrapByCode(dfn, term);
+  const needsCode =
+    ant.firstElementChild && ant.firstElementChild.localName === "code"
```
Comment (sidvishnoi) [inline src/core/link-to-dfn.js]: handle this case in `shouldWrapByCode`?

File: tests/spec/core/xref-spec.js [hunk 2]
```diff
@@ -675,23 +676,29 @@ describe("Core — xref", () => {
...
+      expect(link1b.firstElementChild.localName).toBe("code");
       const vars1 = [...doc.querySelectorAll("#link1 var")];
       expect(vars1.length).toBe(2);
       expect(vars1[0].textContent).toBe("type");
       expect(vars1[1].textContent).toBe("callback");
 
-      const [link2a, link2b] = [...doc.querySelectorAll("#link2 code a")];
+      const [link2a, link2b] = [...doc.querySelectorAll("#link2 a")];
       expect(link2a.href).toBe(expectedLinks.get("ChildNode"));
+      expect(link2a.firstElementChild.localName).toBe("code");
```
Comment (sidvishnoi) [inline tests/spec/core/xref-spec.js]: so these tests (localName ones) are all kinda same.. maybe do them once per type?

**Comment by marcoscaceres** [comment]
> Not sure why we need it though. Some recommendation from tr-design?

The styles/markup is inconsistent with the local ones, so it makes them somewhat confusing... this makes them all the same, but with the ability for us to style them a bit differently as needed.

File: tests/spec/core/xref-spec.js (hunk 2 above)
Comment (marcoscaceres) [reply tests/spec/core/xref-spec.js]: I wanted to keep the original structure we were testing, which structurally was "#link code a" for all of them (now just swapping a and code). I know it's a little redundant, but I think it's ok. More chance to catch any regressions.

**Pull request merged**